RecoEgamma/EgammaElectronProducers/plugins/GEDGsfElectronCoreProducer.cc

```diff
@@ -22,10 +22,12 @@ using namespace reco;
 
 void GEDGsfElectronCoreProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
   edm::ParameterSetDescription desc;
-  GsfElectronCoreBaseProducer::fillDescription(desc);
-  desc.add<edm::InputTag>("GEDEMUnbiased", edm::InputTag("GEDPFCandidates"));
-
-  descriptions.add("produceEcalDrivenGsfElectronCores", desc);
+  desc.add<edm::InputTag>("gsfPfRecTracks", edm::InputTag("pfTrackElec"));
+  desc.add<edm::InputTag>("gsfTracks", edm::InputTag("electronGsfTracks"));
+  desc.add<edm::InputTag>("ctfTracks", edm::InputTag("generalTracks"));
```

Comment:
Having `GsfElectronCoreBaseProducer::fillDescription` for just two entries is not really worth it I think. What do you think about just removing the `GsfElectronCoreBaseProducer::fillDescription` which is unused anyway and then there is no confusion and just 2 duplicate entries? I think having two repeated entries in 2 derived producers `fillDescriptions` is more explicit and easy than trying to put something together with inheritance. I'm not a bit fan of inheritance with producers anyway...

Reply:
There are 4 parameters (by name, if not by value) in common, not 2.
There are 3 derived classes (GsfElectronCoreProducer doesn't have a fillDescriptions yet).
So, the repetition seems excessive.

Reply:
I don't understand how we can have the parameters in the base `fillDescription` that will have different values in the derived plugins without re-introducing the manual cfi files to change them again. Isn't this what we want to avoid? That's why I said we can only put the 2 parameters that do not change in the base `fillDescription`. Maybe you can clarify what you mean? Or just take a look at my most recent commit and tell me if it goes in the right direction? I have now `fillDescriptions` for all the derived producers and therefore the `gsfElectronCores_cfi` configuration is not needed anymore. Everything is now configured with the C++ code and the era modifiers.

Reply:
I meant to mod the base fillDescription to be ```fillDesccription(desc, const string& pfTk, gsfTk)``` and pass appropriate arguments. Recall that only the values of gsfPfRecTracks and gsfTracks change, the parameter names and types themselves are unchanged.

Reply:
I finally see what you mean. Nice we can do it like that! I'll update the last commit.

Reply:
Alright I did now what you suggested and updated the last commit so you can take a look. Now I'm doing local tests. I'll let you know if everything worked.